src/main/java/com/synopsys/integration/alert/provider/polaris/tasks/PolarisProjectSyncTask.java

```diff
@@ -85,76 +87,113 @@ public PolarisProjectSyncTask(final TaskScheduler taskScheduler, final PolarisPr
+            if (null != providerProject) {
+                final Set<String> emailsForProject;
+                try {
+                    emailsForProject = polarisApiHelper.getAllEmailsForProject(projectResource);
+                    projectToUsers.put(providerProject, emailsForProject);
+                } catch (final IntegrationException e) {
+                    logger.error("Failed to get emails for project '{}': ", providerProject.getName(), e.getMessage());
+                }
             }
-        } catch (final IntegrationException e) {
-            logger.error("Failed to get projects from Polaris", e);
         }
-        return projectsFromServer;
+        return projectToUsers;
     }
 
-    private Map<String, ProviderProject> getProjectsFromDatabase() {
-        return projectRepositoryAccessor.findByProviderName(PolarisProvider.COMPONENT_NAME)
-                   .stream()
-                   .collect(Collectors.toMap(ProviderProject::getHref, Function.identity()));
-    }
+    private Map<ProviderProject, Set<PolarisIssueModel>> updateIssuesForProjects(final PolarisApiHelper polarisApiHelper, final Set<ProviderProject> projects, final Map<ProjectV0Resource, List<BranchV0Resource>> projectToBranchMappings) {
+        final Map<ProviderProject, Set<PolarisIssueModel>> projectIssuesMap = new HashMap<>();
+        for (final ProviderProject project : projects) {
+            final String projectHref = project.getHref();
+            final String projectName = project.getName();
+
+            final String projectId;
+            final List<String> branchIds;
+            try {
+                final Optional<ProjectV0Resource> optionalProjectV0Resource = polarisApiHelper.retrieveProjectByHrefOrName(projectToBranchMappings.keySet(), projectHref, projectName);
+                if (optionalProjectV0Resource.isPresent()) {
+                    final ProjectV0Resource ProjectV0Resource = optionalProjectV0Resource.get();
+                    projectId = ProjectV0Resource.getId();
+                    branchIds = polarisApiHelper.getBranchesIdsForProject(projectToBranchMappings, ProjectV0Resource);
+                } else {
+                    continue;
+                }
+            } catch (final IntegrationException e) {
+                logger.error("Problem getting project data from Polaris: {}", projectName, e);
+                continue;
+            }
 
-    private Set<ProviderProject> storeNewProjects(final Map<String, ProviderProject> oldStoredProjects, final Set<ProviderProject> newProjects) {
-        newProjects.forEach(project -> projectRepositoryAccessor.saveProject(PolarisProvider.COMPONENT_NAME, project));
+            final List<PolarisIssueModel> issueModelsFromServer = polarisApiHelper.createIssueModelsForProject(projectId, projectName, branchIds);
+            final Set<PolarisIssueModel> projectIssues = updateIssues(projectHref, issueModelsFromServer);
+            projectIssuesMap.put(project, projectIssues);
```

Comment:
Can these 3 lines be moved into the try if statement and we remove the else continue?